### ISSUE TITLE
Duplicate DNS-01 challenges when using wildcard sans

### ISSUE THREAD

**Issue by lsl** [containous/traefik#3912]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

Trying to setup a domain for use with docker swarm + traefik + wildcard letsencrypt certs over DNS-01

I'm seeing what I suspect are DNS challenges being run in parallel duplicated for each domain.

Depending on which order these run (seems random) the challenges will either fail or succeed. What happens a lot of the time is that it retries >5 times and hits lets encrypt limiting.

Luckily I didn't get hit with rate limiting tonight but I did get a clear log which shows the request auth keys repeating but appearing out of order leading to multiple failures before finally succeeding.

### What did you expect to see?

Expect to see a single set of DNS challenges being run.

### What did you see instead?

Multiple requests being made in jumbled order. While its not shown in the logs below I believe it essentially sets the dns token and then the other thread comes along and overwrites it leading to a failure and restart.

### Output of `traefik version`: 

```
v1.7.0-rc5 (alpine docker)
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

Alpine, docker swarm, GCE debian

Linux instance-1 4.9.0-8-amd64 #1 SMP Debian 4.9.110-3+deb9u4 (2018-08-21) x86_64 GNU/Linux
Docker version 18.06.1-ce, build e68fc7a

```yml
--acme
--acme.acmelogging
--acme.dnschallenge
--acme.dnschallenge.provider=gcloud
--acme.domains='*.example.com,example.com'
--acme.email=snipped@example.com
--acme.entrypoint=https
--acme.ondemand=false
--acme.storage=/lets/acme.json
--checknewversion=false
--defaultentrypoints=https,http
--docker
--docker.domain=example.com
--docker.exposedbydefault=false
--docker.swarmmode
--docker.watch
--entrypoints='Name:http Address::80 Redirect.EntryPoint:https'
--entrypoints='Name:https Address::443 TLS Auth.Forward.Address:http://auth:4181'
--loglevel=INFO
--web
```

### If applicable, please paste the log output in DEBUG level (`--logLevel=DEBUG` switch)

```
2018-09-21 14:43:59 info Using TOML configuration file /traefik.toml
2018-09-21 14:43:59 warning web provider configuration is deprecated, you should use these options : api, rest provider, ping and metrics
2018-09-21 14:43:59 info Traefik version v1.7.0-rc5 built on 2018-09-18_01:50:00PM
2018-09-21 14:43:59 info \nStats collection is disabled.\nHelp us improve Traefik by turning this feature on :)\nMore details on: https://docs.traefik.io/basics/#collected-data\n
2018-09-21 14:43:59 info Preparing server http &{Address::80 TLS:<nil> Redirect:0xc000390d40 Auth:<nil> WhitelistSourceRange:[] WhiteList:<nil> Compress:false ProxyProtocol:<nil> ForwardedHeaders:0xc000477fc0} with readTimeout=0s writeTimeout=0s idleTimeout=3m0s
2018-09-21 14:43:59 info Preparing server https &{Address::443 TLS:0xc0001ba7e0 Redirect:<nil> Auth:0xc00001a780 WhitelistSourceRange:[] WhiteList:<nil> Compress:false ProxyProtocol:<nil> ForwardedHeaders:0xc000300080} with readTimeout=0s writeTimeout=0s idleTimeout=3m0s
2018-09-21 14:43:59 info Starting server on :80
2018-09-21 14:43:59 info Starting server on :443
2018-09-21 14:43:59 info Preparing server traefik &{Address::8080 TLS:<nil> Redirect:<nil> Auth:<nil> WhitelistSourceRange:[] WhiteList:<nil> Compress:false ProxyProtocol:<nil> ForwardedHeaders:0xc000300200} with readTimeout=0s writeTimeout=0s idleTimeout=3m0s
2018-09-21 14:43:59 info Starting provider configuration.ProviderAggregator {}
2018-09-21 14:43:59 info Starting server on :8080
2018-09-21 14:43:59 info Starting provider *docker.Provider {\"Watch\":true,\"Filename\":\"\",\"Constraints\":null,\"Trace\":false,\"TemplateVersion\":2,\"DebugLogGeneratedTemplate\":false,\"Endpoint\":\"unix:///var/run/docker.sock\",\"Domain\":\"example.com\",\"TLS\":null,\"ExposedByDefault\":false,\"UseBindPortIP\":false,\"SwarmMode\":true,\"Network\":\"\"}
2018-09-21 14:43:59 info Starting provider *acme.Provider {\"Email\":\"snipped@example.com\",\"ACMELogging\":true,\"CAServer\":\"https://acme-v02.api.letsencrypt.org/directory\",\"Storage\":\"/lets/acme.json\",\"EntryPoint\":\"https\",\"KeyType\":\"\",\"OnHostRule\":false,\"OnDemand\":false,\"DNSChallenge\":{\"Provider\":\"gcloud\",\"DelayBeforeCheck\":0},\"HTTPChallenge\":null,\"TLSChallenge\":null,\"Domains\":[{\"Main\":\"*.example.com\",\"SANs\":[\"example.com\"]}],\"Store\":{}}
2018-09-21 14:43:59 info Testing certificate renew...
2018-09-21 14:43:59 info The key type is empty. Use default key type 4096.
2018-09-21 14:44:01 info Server configuration reloaded on :443
2018-09-21 14:44:01 info Server configuration reloaded on :8080
2018-09-21 14:44:01 info Server configuration reloaded on :80
2018-09-21 14:44:04 info Server configuration reloaded on :80
2018-09-21 14:44:04 info Server configuration reloaded on :443
2018-09-21 14:44:04 info Server configuration reloaded on :8080
2018-09-21 14:44:15 info Register..
2018-09-21 14:44:15 info legolog: [INFO] acme: Registering account for snipped@example.com
2018-09-21 14:44:15 info legolog: [INFO] [*.example.com, example.com] acme: Obtaining bundled SAN certificate
2018-09-21 14:44:16 info legolog: [INFO] [*.example.com] AuthURL: https://acme-v02.api.letsencrypt.org/acme/authz/ZhZT....
2018-09-21 14:44:16 info legolog: [INFO] [example.com] AuthURL: https://acme-v02.api.letsencrypt.org/acme/authz/a6cE....
2018-09-21 14:44:16 info legolog: [INFO] [example.com] acme: Preparing to solve DNS-01
2018-09-21 14:44:17 info legolog: [INFO] [example.com] acme: Preparing to solve DNS-01
2018-09-21 14:44:18 info legolog: [INFO] [example.com] acme: Trying to solve DNS-01
2018-09-21 14:44:18 info legolog: [INFO] [example.com] Checking DNS record propagation using [127.0.0.11:53]
2018-09-21 14:47:21 error Error obtaining certificate retrying in 656.567213ms
2018-09-21 14:47:22 info legolog: [INFO] [*.example.com, example.com] acme: Obtaining bundled SAN certificate
2018-09-21 14:47:22 info legolog: [INFO] [*.example.com] AuthURL: https://acme-v02.api.letsencrypt.org/acme/authz/a6cE....
2018-09-21 14:47:22 info legolog: [INFO] [example.com] AuthURL: https://acme-v02.api.letsencrypt.org/acme/authz/ZhZT....
2018-09-21 14:47:22 info legolog: [INFO] [example.com] acme: Preparing to solve DNS-01
2018-09-21 14:47:23 info legolog: [INFO] [example.com] acme: Preparing to solve DNS-01
2018-09-21 14:47:24 info legolog: [INFO] [example.com] acme: Trying to solve DNS-01
2018-09-21 14:47:24 info legolog: [INFO] [example.com] Checking DNS record propagation using [127.0.0.11:53]
2018-09-21 14:47:30 info legolog: [INFO] [example.com] The server validated our request
2018-09-21 14:47:30 info legolog: [INFO] [example.com] acme: Trying to solve DNS-01
2018-09-21 14:47:30 info legolog: [INFO] [example.com] Checking DNS record propagation using [127.0.0.11:53]
2018-09-21 14:48:27 error Error obtaining certificate retrying in 404.33751ms
2018-09-21 14:48:27 info legolog: [INFO] [*.example.com, example.com] acme: Obtaining bundled SAN certificate
2018-09-21 14:48:27 info legolog: [INFO] [*.example.com] AuthURL: https://acme-v02.api.letsencrypt.org/acme/authz/a6cE....
2018-09-21 14:48:27 info legolog: [INFO] [example.com] AuthURL: https://acme-v02.api.letsencrypt.org/acme/authz/7R25....
2018-09-21 14:48:27 info legolog: [INFO] [example.com] acme: Authorization already valid; skipping challenge
2018-09-21 14:48:27 info legolog: [INFO] [example.com] acme: Preparing to solve DNS-01
2018-09-21 14:48:29 info legolog: [INFO] [example.com] acme: Trying to solve DNS-01
2018-09-21 14:48:29 info legolog: [INFO] [example.com] Checking DNS record propagation using [127.0.0.11:53]
2018-09-21 14:49:35 info legolog: [INFO] [example.com] The server validated our request
2018-09-21 14:49:35 info legolog: [WARN] Error cleaning up example.com: googlecloud: <nil> 
2018-09-21 14:49:35 info legolog: [INFO] [*.example.com, example.com] acme: Validations succeeded; requesting certificates
2018-09-21 14:49:43 info legolog: [INFO] [*.example.com] Server responded with a certificate.
2018-09-21 14:49:44 info Server configuration reloaded on :80
2018-09-21 14:49:44 info Server configuration reloaded on :443
2018-09-21 14:49:44 info Server configuration reloaded on :8080
```

**Comment by ldez** [comment]
Hi thanks for reporting.

It's already fixed https://github.com/xenolf/lego/pull/643

**Comment by traefiker** [comment]
Closed by #3915.

**Comment by lsl** [comment]
Thanks, hopefully that sorts things.

```
2018-09-21 14:44:15 info legolog: [INFO] [*.example.com, example.com] acme: Obtaining bundled SAN certificate
2018-09-21 14:44:16 info legolog: [INFO] [*.example.com] AuthURL: https://acme-v02.api.letsencrypt.org/acme/authz/ZhZT....
2018-09-21 14:44:16 info legolog: [INFO] [example.com] AuthURL: https://acme-v02.api.letsencrypt.org/acme/authz/a6cE....
2018-09-21 14:47:22 info legolog: [INFO] [*.example.com, example.com] acme: Obtaining bundled SAN certificate
2018-09-21 14:47:22 info legolog: [INFO] [*.example.com] AuthURL: https://acme-v02.api.letsencrypt.org/acme/authz/a6cE....
2018-09-21 14:47:22 info legolog: [INFO] [example.com] AuthURL: https://acme-v02.api.letsencrypt.org/acme/authz/ZhZT....
```

Any idea about the perceived duplication of work? Note the matching and swapping auth url's between pairs. I'm not completely sure how this works so maybe ths is expected.
